www/class/centreonTraps.class.php

```diff
@@ -284,135 +284,216 @@ public function update($traps_id = null)
+        }
 
         $rq .= "`traps_advanced_treatment` = ";
-        isset($ret["traps_advanced_treatment"]) && $ret["traps_advanced_treatment"] != null
-            ? $rq .= "'" . $ret["traps_advanced_treatment"] . "', "
-            : $rq .= "'0', ";
+        if (isset($ret["traps_advanced_treatment"]) && $ret["traps_advanced_treatment"] != null) {
+            $rq .= ':traps_advanced_treatment, ';
+            $retValue[':traps_advanced_treatment'] = $ret["traps_advanced_treatment"];
+        } else {
+            $rq .= "'0', ";
+        }
 
         $rq .= "`traps_comments` = ";
-        isset($ret["traps_comments"]) && $ret["traps_comments"] != null
-            ? $rq .= "'" . $ret["traps_comments"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["traps_comments"]) && $ret["traps_comments"] != null) {
+            $rq .= ':traps_comments, ';
+            $retValue[':traps_comments'] = $ret["traps_comments"];
+        } else {
+            $rq .= "NULL, ";
+        }
 
         $rq .= "`traps_routing_mode` = ";
-        isset($ret["traps_routing_mode"]) && $ret["traps_routing_mode"] != null
-            ? $rq .= "'" . $ret["traps_routing_mode"] . "', "
-            : $rq .= "'0', ";
+        if (isset($ret["traps_routing_mode"]) && $ret["traps_routing_mode"] != null) {
+            $rq .= ':traps_routing_mode, ';
+            $retValue[':traps_routing_mode'] = $ret["traps_routing_mode"];
+        } else {
+            $rq .= "'0', ";
+        }
 
         $rq .= "`traps_routing_value` = ";
-        isset($ret["traps_routing_value"]) && $ret["traps_routing_value"] != null
-            ? $rq .= "'" . $ret["traps_routing_value"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["traps_routing_value"]) && $ret["traps_routing_value"] != null) {
+            $rq .= ':traps_routing_value, ';
+            $retValue[':traps_routing_value'] = $ret["traps_routing_value"];
+        } else {
+            $rq .= "NULL, ";
+        }
 
         $rq .= "`traps_routing_filter_services` = ";
-        isset($ret["traps_routing_filter_services"]) && $ret["traps_routing_filter_services"] != null
-            ? $rq .= "'" . $ret["traps_routing_filter_services"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["traps_routing_filter_services"]) && $ret["traps_routing_filter_services"] != null) {
+            $rq .= ':traps_routing_filter_services, ';
+            $retValue[':traps_routing_filter_services'] = $ret["traps_routing_filter_services"];
+        } else {
+            $rq .= "NULL, ";
+        }
 
         $rq .= "`manufacturer_id` = ";
-        isset($ret["manufacturer_id"]) && $ret["manufacturer_id"] != null
-            ? $rq .= "'" . $ret["manufacturer_id"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["manufacturer_id"]) && $ret["manufacturer_id"] != null) {
+            $rq .= ':manufacturer_id, ';
+            $retValue[':manufacturer_id'] = $ret["manufacturer_id"];
+        } else {
```

Comment:
as this value is mandatory, you're not able to save the form and this case shouldn't exist